### ISSUE TITLE
[Bug] Setting parent and child variable via url does not update child option list. 

### ISSUE THREAD

**Issue by reliam** [grafana/grafana#8951]
Please include this information:
- What Grafana version are you using? 4.4.1 (from packagecloud repo stable/el/6)
- What datasource are you using? Cloudwatch
- What OS are you running grafana on? Amazon Linux (ami-a4c7edb2)
- What did you do? Trying to fill in template values as query strings for a Cloudwatch based dashboard
- What was the expected result? For the proper values to be passed into the template and render data for the specific endpoint
- What happened instead? Nothing renders, and the resource does not appear in the drop-down. The only way to fix is to select a different region and then switch back to the original.

![image](https://user-images.githubusercontent.com/11980918/28731587-5f1e501a-7389-11e7-9b4a-ecc0e583b89b.png)

I'm attempting to create quick links to several resources from another tool, which will use the base url but fill in the query strings with the resource name/region.

**Comment by torkelo** [comment]
Please include a lot more detail on your template variable settings, 

You mention query string in the issue  name, please include more info on what your trying to do with query strings.

**Comment by reliam** [comment]
We're using an internal web application which tracks AWS resources for several accounts. When the pages on the tool are rendered it is generating a link to a template dashboard with the resource metadata being passed in parts of the link.

Like so: https://grafana.fqdn/dashboard/file/elb_metrics.json?orgId=1&var-datasource={{ account.name }}&var-region_name={{ elb.region }}&var-elb={{ elb.name }}

When rendered it would become: https://grafana.fqdn/dashboard/file/elb_metrics.json?orgId=1&var-datasource=Reliam%20Main&var-region_name=us-west-2&var-elb=reliam-prod-grav

Current template values: 
![image](https://user-images.githubusercontent.com/11980918/28735798-ccecfb98-739b-11e7-90c0-db085c951b8e.png)

Currently when I navigate to the link with variable data in query strings no charts are rendered: 
![image](https://user-images.githubusercontent.com/11980918/28735851-167a2ae2-739c-11e7-9a16-7243d0e0c9e1.png)

I have to switch the region_name variable to a different region, and then back to us-west-2 for it to properly poll the Cloudwatch API. It doesn't work on the initial page load.

Let me know if more information is required.

**Comment by torkelo** [comment]
Can you have chrome dev tools open while you load page, and look at the network requests? What queries are sent and what is the aws response?

**Comment by reliam** [comment]
I believe this issue comes down to a bug in my dashboard json. I've updated the refresh period to update on time change on the templates and deleted the "current" fields. 

I believe the "current" objects on each template was using the old data from whenever the dashboard was exported with json output, and subsequent loads would try and render those objects but with the requested datasource, causing a render error.

Once setting the "current" properties empty the graphs are rendering with clickable links with data being passed in the query strings. Closing this issue.
